.github/workflows/tests_entry.yml

```diff
@@ -0,0 +1,95 @@
+# tests_entry.yml
+# Entry point to the test suites
+#
+# - If neither ci:run or ci:ready_to_merge labels are on PR, fail and exit.
+# - If ci:run label is on PR, remove ci:run, run the test scripts.
+# - If ci:ready_to_merge is on the PR and the action is synchronize not triggered
+#   by the mergify[bot] user, remove label and fail job.
+# - If ci:ready_to_merge label is on PR and action is labeled, run the test scripts.
+# 
+# The end result is labeling ci:run or ci:ready_to_merge will run the test scripts,
+# If Mergify merges to the PR, the test scripts will run. If anyone else tries to add 
+# a commit to the PR or merge , the script will fail and ci:ready_to_merge will be
+# removed.
+#
+# This script runs the test scripts directly. Scheduled or manual runs use 
+# run_<scriptname>.yml as the entry point. 
+
+name: Tests Entry Point
+on:
+  pull_request_target:
+    types:
+      - synchronize
+      - labeled
+
+jobs:
+  no-labels:
+    runs-on: ubuntu-latest
+    if: ${{ !(contains(github.event.pull_request.labels.*.name, 'ci:run') ||
+        contains(github.event.pull_request.labels.*.name, 'ci:ready_to_merge')) }}
+    steps:
+      - run: exit 1
+  some-labels:
+    runs-on: ubuntu-latest
+    if: ${{ (contains(github.event.pull_request.labels.*.name, 'ci:run') ||
+        contains(github.event.pull_request.labels.*.name, 'ci:ready_to_merge')) }}
+    steps:
+      - name: remove-ready-to-merge
+        if: ${{ (contains(github.event.pull_request.labels.*.name, 'ci:ready_to_merge') &&
```

Comment:
we could have this be:

```yaml
if: ${{ (contains(github.event.pull_request.labels.*.name, 'ci:ready_to_merge') && !env.SYNC_BY_MERGIFY
```

Reply:
So for all of these take a look at what I just pushed. It was just a matter of moving the logic into the steps instead of at the job level. While that basically reverted the logic line to what we had in the original version, it still provides separation of the tags. I don't think doing more here would help a lot with readability or conceptual clarity.